### DISCUSSION TITLE
recommendation: calendar backend library vkurko/calendar

### DISCUSSION THREAD

**Discussion by mrvnklm** [wdelfuego/nova-calendar#54] [Ideas & Feature Requests]
Hello, as I have some custom needs I decided to work with a javascript calendar library instead of using this package for now, but I wanted to suggest the following library for future versions:

https://github.com/vkurko/calendar

- really lightweight and performant
- many views out of the box
- developer is really active fixing issues 
- interactivity features if needed in the future

If I have the time, I would like to help to integrate it into v2, after the initial release, as I use it in my custom nova tool already.
I might post some pictures of my usecase including event and resource filtering. Might be interesting for https://github.com/wdelfuego/nova-calendar/pull/26

**Comment by wdelfuego** [comment]
Thanks for your post!

That looks like a very nice front-end package 👍 

I'm not sure if this would work with licensing since my package isn't MIT licensed but dual licensed (GPL / proprietary).
Also that calendar comes with a lot of features, something I have tried to avoid in this package in favor of a more plug-and-play approach. 

If the licensing issue can be worked out and the package can be integrated in a way that maintains the default plug-and-play behavior of this package (ie, not requiring any configuration to boot, with sensible defaults), perhaps this could work and be a powerful addition. I can't imagine such a big change/addition could be done within the v2 branch though so I'd say we'd most probably be looking at a v3 or later release or a fork.

**Reply by mrvnklm** [reply]
Okay, so if the licensing issue would be solved, I would suggest to implement basic views / functionality for general purpose usage like now and just add a simple method to extend to advanced functionality by adding some custom javascript code or by importing a file, without adding effort or overhead.

Further mentioning this in the readme, so this would not cause any non package related issues.

**Reply by wdelfuego** [reply]
I'm not sure what you mean by that last sentence wrt the readme.

To be fair, I think the package you mentioned is powerful because it allows customization so it can be used in many ways and in many projects. But this project's goal is to be a Nova calendar, so it doesn't need as many customization features. (Although if they can be effectively hidden and the package made transparently compatible with this package without changing the user or developer experience much, it could still be a powerful addition).

The most valuable feature I see in the package you mentioned that we are still missing here, is event editing by dragging and dropping. I might prefer to add that to the current front-end view rather than switching to a powerful 3rd party engine for the front-end.

In any case, for now, this package will keep being a read-only calendar view since nobody is working on drag and drop nor does it seem to be a much requested feature, so I personally will not be investing time in this for now either. Writing your own back-end code in order to supply Nova resource data to your calendar package might be a better way to go about it than to try to merge with this package. But I'm open to being surprised if anyone's willing to work on an integration.

**Reply by mrvnklm** [reply]
Alright, thats totally fine.

I just thought of this package more as an interface to provide the data needed in a calendar view in a good manner with some degree of customizability and filtering.

As you seem to be maintaining that package in a organized and kinda long term way, I thought handling the data might be enough work, so it might always lack some frontend functionality which could be provided by using some kind of external library, especially when adding more views which also adds more overhead and therefore also bugs at some point.

Nevertheless this is the best Nova Calendar out there, so thank you for your work!
